internal/testsdetection/testsdetection_test.go

```diff
@@ -0,0 +1,87 @@
+package testsdetection_test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testsdetection"
+)
+
+var coverDir string
+
+func TestMustBeTestingInTests(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		require.Nil(t, r, "MustBeTesting should not panic as we are running in tests")
+	}()
+
+	testsdetection.MustBeTesting()
+}
+
+func TestMustBeTestingInProcess(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		integrationtestsTag bool
+
+		wantPanic bool
+	}{
+		"Pass when called in an integration tests build": {integrationtestsTag: true, wantPanic: false},
+
+		"Error (panics) when called in non tests and no integration tests": {integrationtestsTag: false, wantPanic: true},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			temp := t.TempDir()
+			testBinary := filepath.Join(temp, "testbin")
+
+			buildCmd := []string{"build", "-o", testBinary}
+			if tc.integrationtestsTag {
+				buildCmd = append(buildCmd, "-tags=integrationtests")
+			}
+			env := os.Environ()
+			if coverDir != "" {
+				buildCmd = append(buildCmd, "-cover")
+				env = append(env, "GOCOVERDIR="+coverDir)
+			}
+			buildCmd = append(buildCmd, "testdata/binary.go")
+
+			//nolint:gosec // G204 we are in control of the arguments in our tests.
+			out, err := exec.Command("go", buildCmd...).CombinedOutput()
+			require.NoErrorf(t, err, "Setup: Could not build test binary: %s", out)
+
+			// Execute our subprocess
+			//nolint:gosec // G204 we are in control of the arguments in our tests.
+			cmd := exec.Command(testBinary)
+			cmd.Env = env
+			out, err = cmd.CombinedOutput()
+
+			if tc.wantPanic {
+				require.Errorf(t, err, "MustBeTesting should have panicked the subprocess: %s", string(out))
```

Comment:
no need to use `string(out)` if using `%s` already

Reply:
That was my first attempt, but I got a slice of byte printed out here.

Reply:
Weird, I've been using that too recently and it did work.

Reply:
Ahah, found it! The issue is on https://github.com/ubuntu/authd/pull/300/files/531a090e7e64b14d6c4b188b9fa4b508aecedf57..29a6185a9e4c36a2a190aa63bcbf63d2d514619b#diff-c00785a26f57a11b8084dd9fc39d8bab7944c6db4c221fabd2abbd7578461561R71 which is missing `%s` and that was the one I saw when erroring out :)